src/garage/tf/q_functions/discrete_cnn_q_function.py

```diff
@@ -160,14 +162,25 @@ def _initialize(self):
 
     @property
     def q_vals(self):
-        """Q values."""
+        """Return the Q values, the output of the network.
+
+        Return:
+            list[tf.Tensor]: Q values.
+
+        """
         return self.model.networks['default'].outputs
 
     @property
     def input(self):
-        """Input tf.Tensor of the Q-function."""
+        """Get input.
+
+        Return:
+            tf.Tensor: QFunction Input.
+
+        """
         return self.model.networks['default'].input
 
+    # pylint: disable=arguments-differ
```

Comment:
reason? doesn't seem like this should different from the ABC

Reply:
it does, sometimes it takes one argument (state), sometimes two (state, action)

Reply:
that should not be the case for a Q-function -- it should always be conditioned on (s,a)

who is conditioning a Q-function on just state?

Reply:
discrete q-function only takes state in garage, and output all the Q values for all actions... sounds like this implementation is not right?

Reply:
if that's the case, can we fix that in another PR?